packages/runtime-common/realm.ts

```diff
@@ -1783,6 +1789,31 @@ export class Realm {
     };
   }
 
+  private async visibility(): Promise<RealmVisibility> {
+    if (!this.visibilityCache) {
+      let permissions = await fetchUserPermissions(
+        this.#dbAdapter,
+        new URL(this.url),
+      );
+
+      let usernames = Object.keys(permissions);
+      if (usernames.includes('*')) {
+        this.visibilityCache = 'public';
+      } else if (usernames.includes('users')) {
+        this.visibilityCache = 'shared';
+        // Defaultly a realm will have two users:
+        // 1. The user who created the realm
+        // 2. The worker assigned to it
+      } else if (usernames.length > 2) {
```

Comment:
i didn't even think about this, but that's a good observation. maybe a better way is to just filter out the realm user from the usernames. The realm users always start with `realm/`. and then we can just change usernames.length to 1.